### PR TITLE
AO3-5717 Fix expected chapter count lower than actual count

### PR DESCRIPTION
**Pull request by luis-pabon-tf**
# Pull Request Checklist

## Issue

https://otwarchive.atlassian.net/browse/AO3-5717

## Purpose

Uses an existing virtual attribute so that the chapter count safeguards happen even if the chapter count provided is wrong.

## Testing Instructions

In the JIRA ticket.

## Credit

What name and pronouns should we use to credit you in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?

bird - (they)

### CODE REVIEW

**Review by redsummernight — COMMENTED** [review]
Can you add a test for this behavior of `wip_length`? In spec/models/work_spec.rb, we can have a block for `describe "#wip_length="` and check that setting the attribute to e.g. 2 for a work with 1 chapter will end up with both `expected_number_of_chapters` and `wip_length` at 1.

File: spec/models/work_spec.rb [hunk 1]
```diff
@@ -464,6 +464,41 @@
     end
   end
 
+  describe "#wip_length_equals" do
```
Comment (redsummernight) [inline spec/models/work_spec.rb:467]: Can you change this to `#wip_length=` to match the method name?

File: spec/models/work_spec.rb [hunk 2]
```diff
@@ -464,6 +464,41 @@
...
+    it "updating chapter count via expected_number_of_chapters ignores any safeguard logic" do
```
Comment (redsummernight) [inline spec/models/work_spec.rb:468]: I think we can skip this test, we don't have to check that the default `expected_number_of_chapters` setter works as expected.

**Comment by tickinginstant** [comment]
Would you mind removing `expected_number_of_chapters` from the `work_params` function in `WorksController`, as well?
https://github.com/otwcode/otwarchive/blob/72b975cf5ed90ce28cf212e84420aaccea871372/app/controllers/works_controller.rb#L893-L896
Otherwise someone would still be able to manipulate the HTML to get an invalid chapter count.

**Review by redsummernight — APPROVED** [review]
Looks good, thanks!

**Pull request merged**